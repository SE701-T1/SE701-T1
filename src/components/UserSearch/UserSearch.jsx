import React, { useState } from 'react';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import SearchIcon from '@mui/icons-material/Search';
import InputAdornment from '@mui/material/InputAdornment';
import { styled } from '@mui/material/styles';
import styles from './UserSearch.module.css';

// searchbox styling
const SearchBox = styled(TextField)(() => ({
  width: '100%',
  '& fieldset': {
    borderRadius: '25px',
    backgroundColor: '#FAFAFC',
    zIndex: -1,
  },

  '& .MuiInputBase-input': {
    width: '100%',
    color: 'black',
  },

  '& .MuiOutlinedInput-root': {
    '&:hover fieldset': {
      borderColor: '#BFBFBF',
    },
    '&.Mui-focused fieldset': {
      borderColor: '#BFBFBF',
    },
  },
}));

/**
 * This component takes in a function as a prop that should handle the behaviour of a user clicking on the search icon.
 *
 */
export default function UserSearch({ onSearch }) {
  const [searchInput, setSearchInput] = useState('');

  const handleSearching = (event) => {
    const searchValue = event.target.value;
    setSearchInput(searchValue);
    onSearch(searchValue);
  };

  return (
    <Box className={styles.root} component="form" noValidate autoComplete="off">
      <SearchBox
        onKeyPress={(e) => e.key === 'Enter' && e.preventDefault()}
        id="outlined-basic"
        variant="outlined"
        placeholder="Type a name"
        value={searchInput}
        onChange={handleSearching}
        size="small"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <SearchIcon data-testid="searchIcon" />
            </InputAdornment>
          ),
        }}
      />
    </Box>
  );
}
