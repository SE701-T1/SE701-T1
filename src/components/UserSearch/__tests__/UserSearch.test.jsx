import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import UserSearch from '../UserSearch';

// unit tests
describe('User search test', () => {
  it('should display the placeholder text', () => {
    render(<UserSearch />);

    expect(screen.getByPlaceholderText('Type a name')).toBeInTheDocument();
  });

  it('should display the search icon', () => {
    render(<UserSearch />);

    expect(screen.getByTestId('searchIcon'));
  });

  it('check that handle submit works after clicking search icon', () => {
    const mockOnClick = jest.fn();

    render(<UserSearch submitSearch={mockOnClick()} />);

    const clickIndicator = screen.getByTestId('searchIcon');

    fireEvent.click(clickIndicator);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  // snapshot test
  it('produces correct snapshot of the UserSearch', () => {
    const tree = renderer.create(<UserSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
