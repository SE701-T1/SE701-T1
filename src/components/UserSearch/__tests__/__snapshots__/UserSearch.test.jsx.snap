// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User search test produces correct snapshot of the UserSearch 1`] = `
<form
  autoComplete="off"
  className="root MuiBox-root css-0"
  noValidate={true}
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1rgxkzw-MuiFormControl-root-MuiTextField-root"
    onKeyPress={[Function]}
  >
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="searchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type a name"
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-hdw1oc"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;
